WPMedia/fusion-news-theme-blocks
# TMEDIA-19 - Proof of Concept integration on blocks (wip)

## Description

Using the components from #867 to update a couple of blocks for a proof of concept.

~~Note: If top table list is used without a heading before it as it's wrapped with HeadingSection it's start the headings at 3. Need to look into this a little more.~~

Fixed by making context start at 1

**UPDATE**

Created a new header block that is a chain block to ensure heading hierarchy is increased for children. Any block that is a child of the new header-chain-block will ensure their heading levels are increased.


## Jira Ticket
- [TMEDIA-19](https://arcpublishing.atlassian.net/browse/TMEDIA-19)

## Test Steps

Required feature pack env settings - 

`FUSION_RELEASE=2.7.3-themes-linking.1`
`ENGINE_SDK_DIST_TAG=canary`

Update blocks.json to have new block included `@wpmedia/header-chain-block@local`


1. Checkout this branch `git checkout TMEDIA-19-poc-on-blocks && git remote update --prune origin && git reset --hard origin/TMEDIA-19-poc-on-blocks`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/shared-styles,@wpmedia/numbered-list-block,@wpmedia/header-block,@wpmedia/top-table-list-block,@wpmedia/simple-list-block,@wpmedia/header-chain-block`
3. Use editor to add header block, top table list and numbered list blocks to a page
4. Verify headings follow sequential order

## Effect Of Changes
### Before

Headings are not following sequential order on page or within blocks

<img width="1633" alt="Microsoft Edge-2021-05-13-13-26 53" src="https://user-images.githubusercontent.com/868127/118244492-7f13d680-b497-11eb-9b3e-4f6fe105ffc1.png">


### After

Headings increase as needed when used within a HeadingSection

<img width="1633" alt="Microsoft Edge-2021-05-13-13-24 49" src="https://user-images.githubusercontent.com/868127/118244517-863ae480-b497-11eb-9276-e3cef1aaf21a.png">


New Header Chain Block
<img width="1298" alt="Microsoft Edge-2021-05-17-13-20 17" src="https://user-images.githubusercontent.com/868127/118487472-a9bd9380-b712-11eb-88b5-26a209e38325.png">

Published page using header chain block

<img width="1019" alt="Microsoft Edge-2021-05-17-13-21 07" src="https://user-images.githubusercontent.com/868127/118487568-c2c64480-b712-11eb-91b4-57736d749b57.png">



## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
